src/vm/readytoruninfo.cpp

```diff
@@ -890,6 +890,84 @@ PCODE ReadyToRunInfo::MethodIterator::GetMethodStartAddress()
     return ret;
 }
 
+BOOL ReadyToRunInfo::GenericMethodIterator::Next()
+{
+    m_current = m_enum.GetNext();
```

Comment:
I think if you modified the `GenericMethodIterator` to use a `NativeFormat::NativeHashtable::Enumerator` instead, you could modify this line to be 
```c++
m_enum.GetNext(m_current);
```
and achieve the same thing without the additional iterator class.